docs/extending/custom_bulk_actions.rst

```diff
@@ -0,0 +1,275 @@
+returns ``False`` will be available in the context under the key ``'items_with_no_access'``.
+
+.. code-block:: python
+
+    def check_perm(self, obj):
+        return obj.has_perm('some_perm')  # returns True or False
+
+
+The success message shown on the admin can be customised by overriding the ``get_success_message`` method.
+
+.. code-block:: python
+
+    def get_success_message(self, num_parent_objects, num_child_objects):
+        return _("{} objects, including {} child objects have been updated".format(num_parent_objects, num_child_objects))
+
+
+
+Adding bulk actions to the page explorer
+----------------------------------------
+
+When creating a custom bulk action class for pages, subclass from ``wagtail.admin.views.pages.bulk_actions.page_bulk_action.PageBulkAction``
+instead of ``wagtail.admin.views.bulk_action.BulkAction``
+
+Basic example
+~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    from wagtail.admin.views.pages.bulk_actions.page_bulk_action import PageBulkAction
+    from wagtail import hooks
+
+
+    @hooks.register('register_bulk_action')
+    class CustomPageBulkAction(PageBulkAction):
+        ...
+
+
+
+Adding bulk actions to the Images listing
+-----------------------------------------
+
+When creating a custom bulk action class for images, subclass from ``wagtail.images.views.bulk_actions.image_bulk_action.ImageBulkAction``
+instead of ``wagtail.admin.views.bulk_action.BulkAction``
+
+Basic example
+~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    from wagtail.images.views.bulk_actions.image_bulk_action import ImageBulkAction
+    from wagtail import hooks
+
+
+    @hooks.register('register_bulk_action')
+    class CustomImageBulkAction(ImageBulkAction):
+        ...
+
+
+
+Adding bulk actions to the documents listing
+--------------------------------------------
+
+When creating a custom bulk action class for documents, subclass from ``wagtail.documents.views.bulk_actions.document_bulk_action.DocumentBulkAction``
+instead of ``wagtail.admin.views.bulk_action.BulkAction``
+
+Basic example
+~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    from wagtail.documents.views.bulk_actions.document_bulk_action import DocumentBulkAction
+    from wagtail import hooks
+
+
+    @hooks.register('register_bulk_action')
+    class CustomDocumentBulkAction(DocumentBulkAction):
+        ...
+
+
+
+Adding bulk actions to the user listing
+---------------------------------------
+
+When creating a custom bulk action class for users, subclass from ``wagtail.users.views.bulk_actions.user_bulk_action.UserBulkAction``
+instead of ``wagtail.admin.views.bulk_action.BulkAction``
+
+Basic example
+~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    from wagtail.users.views.bulk_actions.user_bulk_action import UserBulkAction
+    from wagtail import hooks
+
+
+    @hooks.register('register_bulk_action')
+    class CustomUserBulkAction(UserBulkAction):
+        ...
+
+
+Adding bulk actions to the snippets listing
```

Comment:
Just a heads up @KIRA009  @kaedroho this change accidentally created a duplicate bulk actions documentation page.

I think it was done around the same time as 0be024e024140abc43e9f1cac2357ebe8e717ffc 

I will get a fix up shortly.

Reply:
https://github.com/wagtail/wagtail/pull/8876